server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -135,6 +138,34 @@ public ExtensionsManager(Path extensionsPath) throws IOException {
         // will be initialized in initializeServicesAndRestHandler which is called after the Node is initialized
         this.transportService = null;
         this.clusterService = null;
+        this.identityService = null;
+        this.client = null;
+        this.extensionTransportActionsHandler = null;
+
+        /*
+         * Now Discover extensions
+         */
+        discover();
+
+    }
+
+    /**
+     * Instantiate a new ExtensionsManager object to handle requests and responses from extensions. This is called during Node bootstrap.
+     *
+     * @param extensionsPath  Path to a directory containing extensions.
+     * @param identityService  Identity Service
+     * @throws IOException  If the extensions discovery file is not properly retrieved.
+     */
+    public ExtensionsManager(Path extensionsPath, IdentityService identityService) throws IOException {
+        logger.info("ExtensionsManager initialized");
+        this.extensionsPath = extensionsPath;
+        this.initializedExtensions = new HashMap<String, DiscoveryExtensionNode>();
+        this.extensionIdMap = new HashMap<String, DiscoveryExtensionNode>();
```

Comment:
I am not sure if this was like this already, but it seems expensive to create a second map here with almost identical entries. Maybe we can add a flag to the ExtensionNode class or something similar?

Reply:
In this version of the code I had 2 constructors and only one or the other would be called. One constructor would take in an IdentityService which was needed in the `discover` function to get the settings registered by the `IdentityService`.

I have since changed that to only have a single constructor and introduce a seconds argument of `Set<Setting> additionalSettings`. Now inside of `Node.java` the settings registered by the identity plugin will be read and passed into the constructor of `ExtensionsManager`.